litert/vendors/qualcomm/core/builders/matmul_op_builder.cc

```diff
@@ -19,14 +23,37 @@ std::vector<OpWrapper> BuildMatmulOp(
     const bool adj_y) {
   std::vector<OpWrapper> res;
 
+  auto dequant_input_0 = tensor_pool.CreateNativeTensor(
+      QNN_DATATYPE_FLOAT_32, {}, inputs[kZero].get().GetDims());
+
+  auto dequant_input_1 = tensor_pool.CreateNativeTensor(
+      QNN_DATATYPE_FLOAT_32, {}, inputs[kOne].get().GetDims());
+
+  auto dequant_matmul_output = tensor_pool.CreateNativeTensor(
+      QNN_DATATYPE_FLOAT_32, {}, outputs[kZero].get().GetDims());
```

Comment:
Need to use `auto&` here, otherwise it will copy and become local variables which will become invalid afterward.

```suggestion
  auto& dequant_input_0 = tensor_pool.CreateNativeTensor(
      QNN_DATATYPE_FLOAT_32, {}, inputs[kZero].get().GetDims());

  auto& dequant_input_1 = tensor_pool.CreateNativeTensor(
      QNN_DATATYPE_FLOAT_32, {}, inputs[kOne].get().GetDims());

  auto& dequant_matmul_output = tensor_pool.CreateNativeTensor(
      QNN_DATATYPE_FLOAT_32, {}, outputs[kZero].get().GetDims());
```